File: tang/src/main/java/com/microsoft/tang/implementation/types/PackageNodeImpl.java
package com.microsoft.tang.implementation.types;

import com.microsoft.tang.types.Node;
import com.microsoft.tang.types.PackageNode;

public class PackageNodeImpl extends AbstractNode implements PackageNode {
  public PackageNodeImpl(Node parent, String name, String fullName) {
    super(parent, name, fullName);
  }
  /**
   * Unlike normal nodes, the root node needs to take the package name of its
   * children into account.  Therefore, we use the full name as the key when
   * we insert nodes into the root.
   */
  @Override
  public void put(Node n) {
    super.children.put(n.getFullName(), n);
  }

}

File: tang/src/main/cs/Tang/Implementations/PackageNodeImpl.cs
using System;
using Com.Microsoft.Tang.Types;

namespace Com.Microsoft.Tang.Implementations
{
    public class PackageNodeImpl : AbstractNode, IPackageNode
    {
        public PackageNodeImpl(INode parent, String name, String fullName) : 
            base(parent, name, fullName)
        {
        }

        public PackageNodeImpl()
            : base(null, "", "[root node]")
        {
        }

        /**
        * Unlike normal nodes, the root node needs to take the package name of its
        * children into account.  Therefore, we use the full name as the key when
        * we insert nodes into the root.
        */
        public void put(INode n) {
            children.Add(n.GetFullName(), n);
        }
    }
}
